lib/facebook_app_events.dart

```diff
@@ -165,7 +164,7 @@ class FacebookAppEvents {
   /// Parameter [registrationMethod] is used to specify the method the user has
   /// used to register for the app, e.g. "Facebook", "email", "Google", etc.
   /// See: https://developers.facebook.com/docs/reference/androidsdk/current/facebook/com/facebook/appevents/appeventsconstants.html/#eventnamecompletedregistration
-  Future<void> logCompletedRegistration({String registrationMethod}) {
+  Future<void> logCompletedRegistration({String? registrationMethod}) {
```

Comment:
I think `registrationMethod` is required

```java
/**
 * This function assumes logger is an instance of AppEventsLogger and has been
 * created using AppEventsLogger.newLogger() call.
 */
public void logCompleteRegistrationEvent (String registrationMethod) {
    Bundle params = new Bundle();
    params.putString(AppEventsConstants.EVENT_PARAM_REGISTRATION_METHOD, registrationMethod);
    logger.logEvent(AppEventsConstants.EVENT_NAME_COMPLETED_REGISTRATION, params);
}
```

Reply:
I just had a look at the source code of one of my app currently on the market, and I am not using the default parameter name for the signup method, which leads me to think that it is not really required. (I was using "method" instead of "fb_registration_method" and it worked for the past few months)

Reply:
Before null saftey it was a developer interpretation choice... but now with null safety it will be comedictated by the design of the plugin. So it would be good to find clear documentation from Facebook SDK definition.

Reply:
Hi @DennisAlund, I found out that an event's parameters are always optional! Of course, this is only true for the events being logged through the **logEvent** method. (https://developers.facebook.com/docs/app-events/overview)

So the registrationMethod parameter should not be required since that would mean adding a constraint that Facebook does not require.

Reply:
@GaspardMerten agree, here you can see the official documentation by Facebook in Swift

https://developers.facebook.com/docs/swift/reference/structs/appevent.html

```swift
public static func completedRegistration(registrationMethod:String?=nil, valueToSum:Double?=nil, extraParameters:ParametersDictionary=[:])
```

Method Name | Description
-- | --
registrationMethod | Optional registration method used.
valueToSum | Optional value to sum.
extraParameters | Optional dictionary of extra parameters.